base/dates/arithmetic.jl

```diff
@@ -68,5 +68,10 @@ end
 
 (.+){T<:TimeType}(x::AbstractArray{T}, y::Period) = reshape(T[i + y for i in x], size(x))
 (.-){T<:TimeType}(x::AbstractArray{T}, y::Period) = reshape(T[i - y for i in x], size(x))
-(.+){T<:TimeType}(y::Period, x::AbstractArray{T}) = x .+ y
-(.-){T<:TimeType}(y::Period, x::AbstractArray{T}) = x .- y
+(.+)(y::Period, x::AbstractArray{TimeType}) = x .+ y
```

Comment:
this changes the meaning since `TimeType` is abstract, you probably won't have arrays of them very often